compiler/src/org.graalvm.compiler.asm.test/src/org/graalvm/compiler/asm/test/AssemblerTest.java

```diff
@@ -28,6 +28,7 @@
 
 import java.lang.reflect.Method;
 
+import jdk.vm.ci.aarch64.AArch64;
```

Comment:
We don't want to make this class depend on AArch64 or HotSpot specific semantics. I suggest moving the `AArch64BitCountAssemblerTest` class to be a sibling of `AArch64UncompressPointerTest`.

Reply:
You should then be able to use `org.graalvm.compiler.hotspot.aarch64.AArch64HotSpotBackend.emitInvalidatePlaceholder(CompilationResultBuilder, AArch64MacroAssembler)` to avoid duplicating this `nop` inserting logic.

Reply:
I can still use `masm.nop()` without relocating this source file. The purpose of adding this tricky logic in `AssemblerTest` is to avoid adding `nop()` in each test case and causing potential failures in the future. But I can change my fix if you don't like it.